PythonVisualizations/SimpleSorting.py

```diff
@@ -450,9 +463,9 @@ def insertionSort(self):
             self.highlightCodeTags('temp_assignment', callEnviron)
             if tempVal:
                 tempVal, _ = self.assignToTemp(
-                    outer, varName="temp", existing=label)
+                    outer, callEnviron, varName="temp", existing=label)
             else:
-                tempVal, label = self.assignToTemp(outer, varName="temp")
+                tempVal, label = self.assignToTemp(outer, callEnviron, varName="temp")
                 callEnviron.add(label)
             callEnviron.add(tempVal.display_shape)
             callEnviron.add(tempVal.display_val)

```

Comment:
The changes to `assignToTemp()` make it responsible for adding the temporary canvas items (`shape` and `val`) to the `callEnviron`.  It then removes them in lines 104-105.  The caller, `insertionSort()`, immediately puts them right back in the `callEnviron` in lines 470-471.  I think it would be better to drop lines 104-105 and 470-471, and just leave those items in the `callEnviron`.  It's now a "side effect" of  `assignToTemp()` that it adds some items to the `callEnviron` that will need to be cleaned up.
Note `assignFromTemp()` doesn't add any new temporary items, so it doesn't need to add anything  to the `callEnviron`.  When it's done, however, `insertionSort()` removes the temporary items from the the `callEnviron` so they stay on the canvas.  It might be more "symmetric" if `assignFromTemp()` also took `callEnviron` as an argument and updated it once the items are placed into `self.list`.

Reply:
@lilypolonetsky and I made changes to fix the redundancy issue, we removed lines 104-105 and 470-471. We left line 469 because the `temp` label would otherwise not get cleaned up when the animation is stopped.